### PR TITLE
change how proto acquires fsharp.core

### PR DESCRIPTION
**Pull request by KevinRansom**

### CODE REVIEW

File: src/fsharp/FSharp.Compiler.Interactive.Settings/FSharp.Compiler.Interactive.Settings.fsproj [hunk 1]
```diff
@@ -27,8 +27,21 @@
     <Compile Include="..\fsiaux.fs" />
   </ItemGroup>
 
-  <ItemGroup>
-    <ProjectReference Include="..\FSharp.Core\FSharp.Core.fsproj" />
+  <!--  Non-proto Build use project reference -->
```
Comment (dsyme) [inline src/fsharp/FSharp.Compiler.Interactive.Settings/FSharp.Compiler.Interactive.Settings.fsproj]: Factor this out into `Directory.Build.targets` somehow

**Comment by KevinRansom** [comment]
@dsyme,  Okay, this PR now fails, because the Proto compiler using the FSharp.Core from build from source can not build the end-product compiler.  That is because the FSharp.Core from build from source does not have ValueOption.

However, if the LKG builds a proto FSharp.Core from the source code in our repo.  Everything works great and as expected.  This got me to really think about things, and now I realize that I don't really understand your problem at all.

The process currently is and should be:
**LKG** builds **Proto** builds **Product**.

Where we are talking about all of the components that make up the compiler:
FSharp.Build … FSharp.Compiler.Private … fsc etc …

It is not clear to me why we need to reuse and old FSharp.Core in order to make this work.

In short:  what are the specific problems that stops the LKG from being able to build FSharp.Core.

Thanks

Kevin

**Comment by dsyme** [comment]
> In short: what are the specific problems that stops the LKG from being able to build FSharp.Core.

OK I see, yes, using the LKG FSharp.Core is a problem.

There are two issues I facec:

1. new FSharp.Core may contain constructs not recognised by LKG, such as nullness annotations.  We can workaround that via `#if BUILD_FROM_SOURCE` - it's a PITA but we can do it.

2. there is a bug in the LKG preventing the compilation of overloads that differ only by return type (which is needed for `op_Explicit` operators, related to #6345) - that bug is only fixed by #6345 itself.  

I'll try to workaround (2) using `#if BUILD_FROM_SOURCE` in FSharp.Core - but ideally we just wouldn't need to do this, e.g. if Proto could somehow use FSharp.Core from a package (I've been using 4.6.2). But I understand if we can't.

**Comment by dsyme** [comment]
OK I looked into (2) in the context of #6345 - I can workaround using `BUILDING_WITH_LKG`.

The problem is that the LKG (and all prior versions of F#) have a limitation where no overloaded method in FSharp.Core can be inlined.  But #6345 introduces a whole bunch of these methods (and also fixes that limitation).

**Comment by dsyme** [comment]
@KevinRansom Thank you for trying with this.  We can close it - I've been able to `#if` my way out of trouble
